### ISSUE TITLE
False positive when invoking abstract method of a enum

### ISSUE THREAD

**Issue by ky-gog** [typetools/checker-framework#5165]
When referring a enum value with abstract method, checker framework use annotation on method declaration instead of method override.

# Command

`$ checker/bin/javac -processor nullness EnumExplicit.java`

# Input
File: `./EnumExplicit.java`
```java
class EnumExplicit {

    public static enum EnumWithMethod {
        VALUE {
            @Override
            public void call(@Nullable String string) {
                // Null string is acceptable in this function.
            }
        };

        public abstract void call(String string);
    }

    public static void main(String[] args) {
        EnumWithMethod.VALUE.call(null);
    }

}
```

# Output
```
EnumExplicit.java:18: error: [argument] incompatible argument for parameter string of call.
        EnumWithMethod.VALUE.call(null);
                                  ^
  found   : null (NullType)
  required: @Initialized @NonNull String
1 error
```

version output:
```
javac 11.0.13
git.branch=master
git.build.host=buffalo.cs.washington.edu
git.build.user.email=
git.build.user.name=Michael Ernst
git.build.version=3.22.2
git.closest.tag.commit.count=6005
git.closest.tag.name=after-reformatting
git.commit.id=c67a31b6f1a7e01ec6347e032d85bed76bdebdc1
git.commit.id.abbrev=c67a31b
git.commit.id.describe=after-reformatting-6005-gc67a31b-dirty
git.commit.message.full=Add citations\n
git.commit.message.short=Add citations
git.commit.time=2022-06-14T10\:51\:59-0700
git.commit.user.email=mernst@cs.washington.edu
git.commit.user.name=Michael Ernst
git.dirty=true
git.remote.origin.url=/scratch/mernst/cf-release/interm/checker-framework
git.tags=
git.total.commit.count=17419
Note: Checker Framework 3.22.2
```

# Expectation

Checkers framework should allow function call in example instead of reporting an error.

When referring a specific enum value, check framework should use annotations on override methods of that value, instead of annotations on abstract method that is overridden.

**Comment by mernst** [comment]
Added test case as file `checker/tests/nullness/FenumExplicit.java`.
